tests/integration/security/healthcheck/mtls_healthcheck_test.go

```diff
@@ -19,50 +19,26 @@ package healthcheck
 import (
 	"testing"
 
-	"istio.io/istio/pkg/test/framework/components/galley"
-	"istio.io/istio/pkg/test/framework/label"
-
-	"istio.io/istio/pkg/test/framework/components/apps"
-	"istio.io/istio/pkg/test/framework/components/deployment"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
 	"istio.io/istio/pkg/test/framework/components/environment"
 	"istio.io/istio/pkg/test/framework/components/namespace"
-	"istio.io/istio/pkg/test/framework/components/pilot"
-
-	"istio.io/istio/pkg/test/framework"
-	"istio.io/istio/pkg/test/framework/components/istio"
-)
-
-var (
-	ist istio.Instance
 )
 
-func TestMain(m *testing.M) {
-	framework.NewSuite("mtls_healthcheck", m).
-		RequireEnvironment(environment.Kube).
-		Label(label.CustomSetup).
-		SetupOnEnv(environment.Kube, istio.Setup(&ist, setupConfig)).
-		Run()
-}
-
-func setupConfig(cfg *istio.Config) {
-	if cfg == nil {
-		return
-	}
-	cfg.Values["sidecarInjectorWebhook.rewriteAppHTTPProbe"] = "true"
-}
-
 // TestMtlsHealthCheck verifies Kubernetes HTTP health check can work when mTLS
 // is enabled.
 // Currently this test can only pass on Prow with a real GKE cluster, and fail
 // on CircleCI with a Minikube. For more details, see https://github.com/istio/istio/issues/12754.
 func TestMtlsHealthCheck(t *testing.T) {
-	ctx := framework.NewContext(t)
-	defer ctx.Done(t)
-	ctx.RequireOrSkip(t, environment.Kube)
+	framework.NewTest(t).
```

Comment:
maybe not part of the pr. but why we need to switch to this new layout?

- separate main
- nesting test as NewTest.Run(). it seems the test is responsible for deleting policy itself?

Reply:
It's generally preferred (although not required) to use this new pattern since it gives more control to the test framework. With `framework.NewTest` it's easy to add labels directly (e.g. "flaky") which can automatically switch on/off behavior on CI.